airbyte-webapp/src/core/form/schemaToFormBlock.ts

```diff
@@ -37,20 +38,49 @@ export const jsonSchemaToUiWidget = (
   }
 
   if (jsonSchema.oneOf?.length && jsonSchema.oneOf.length > 0) {
-    const conditions = Object.fromEntries(
-      jsonSchema.oneOf.map((condition) => {
-        if (typeof condition === "boolean") {
-          return [];
-        }
-        return [condition.title, jsonSchemaToUiWidget({ ...condition, type: jsonSchema.type }, key, path)];
-      })
+    let possibleConditionSelectionKeys = null as Set<string> | null;
+    const conditions = jsonSchema.oneOf.flatMap((condition) => {
+      if (typeof condition === "boolean") {
+        throw new FormBuildError("Spec uses oneOf without using object types for all conditions");
```

Comment:
You are 100% right, fixed

Reply:
I think there are two kinds of errors thrown from our code:
* Our fault - unexpected errors which are 100% on us because there is a bug in the code
* User fault - errors that can happen due to user input

I think the first kind doesn't need to be i18n-ified, it will most likely anyway not tell the user anything useful, it's just relevant for devs to figure out what's going on.

The second kind should be i18n-ified because ideally the user can fix it themselves.

I originally though these errors would be of the first kind, but on second thought they can also be seen as user errors, especially with the push to allow custom connectors. So I went ahead and translated them. `FormBuildError` is now only meant for errors which are caused by a json schema not following the documented rules. I changed one place where it was thrown back to `Error` because it can only be caused by a bug on our side.